app/models/user.rb

```diff
@@ -18,6 +20,36 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  def authenticated? remember_token
+    return false unless remember_token
+
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
+      BCrypt::Password.create string, cost: cost
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_column remember_digest: User.digest(remember_token)
```

Comment:
```suggestion
    update_columns remember_digest: User.digest(remember_token)
```
code ni em chạy chưa??

Reply:
em quên không check database nên không nhận ra lỗi chính tả ạ@@